source/dubregistry/repositories/bitbucket.d

```diff
@@ -120,13 +124,22 @@ class BitbucketRepository : Repository {
 		if( ver.startsWith("~") ) ver = ver[1 .. $];
 		else ver = ver;
 		auto venc = () @trusted { return encodeComponent(ver); } ();
-		return "https://bitbucket.org/"~m_owner~"/"~m_project~"/get/"~venc~".zip";
+		const url = "https://bitbucket.org/"~m_owner~"/"~m_project~"/get/"~venc~".zip";  
+		if (m_authUser.length) return "https://"~m_authUser~":"~m_authPassword~"@"~url["https://".length..$];
+		return url;
 	}
 
 	void download(string ver, scope void delegate(scope InputStream) @safe del)
 	{
 		downloadCached(getDownloadUrl(ver), del);
 	}
+
+	private string getAPIURLPrefix()
+	{
+		import std.uri : encodeComponent;
+		if (m_authUser.length) return "https://"~m_authUser~":"~m_authPassword~"@api.bitbucket.org";
```

Comment:
I think you actually want to use encodeComponent in here and not just leave the import there. Also do that in the function above.

Reply:
OK. Might wanna check the github provider for the same stuff.